server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -483,8 +484,11 @@ public NettyMetrics getNettyMetrics() {
   }
 
   public boolean isDecommissioning() {
-    return ServerStatus.DECOMMISSIONING.equals(serverStatus.get())
-        || ServerStatus.DECOMMISSIONED.equals(serverStatus.get());
+    return ServerStatus.DECOMMISSIONING.equals(serverStatus.get());
+  }
```

Comment:
better reflects the name of the method, no need to bind the `while` condition in `waitDecommissionFinish` on `DECOMMISSIONED` state.